zoneminder/zoneminder-event-cleanup.sh

```diff
@@ -4,23 +4,31 @@
 # Purpose:	Clean up old Zoneminder events but exclude archived events
 # Authors:	Claudio Kuenzler (2018)
 # Doc:		https://www.claudiokuenzler.com/blog/814/how-to-manually-clean-up-delete-zoneminder-events
-# History:	
-# 2018-12-14 First version
+# History:
+# 2020-08-06 changed for zoneminder 1.24+ (@Brawn1)
+# 2020-07-24 added Docker Mysql (@Brawn1)
 # 2018-12-18 DB Name variable in first query
+# 2018-12-14 First version
 ###########################################################################
 # User variables
-olderthan=2 # Defines the minimum age in days of the events to be deleted
-zmcache=/var/cache/zoneminder/events # Defines the path where zm stores events
+olderthan=7 # Defines the minimum age in days of the events to be deleted
+zmcache=/var/lib/docker/volumes/zoneminder_zmcache/_data/events # Defines the path where zm stores events
 mysqlhost=localhost # Defines the MySQL host for the zm database
 mysqldb=zm # Defines the MySQL database name used by zm
-mysqluser=zmuser # Defines a MySQL user to connect to the database
-mysqlpass=secret # Defines the password for the MySQL user
+mysqluser=zoneminder # Defines a MySQL user to connect to the database
+mysqlpass=zm-mysql-password # Defines the password for the MySQL user
+docker_mysql="" # if mysql running as docker container, add the mysql container name (example: docker_mysql=zoneminder_mysql_1), or leafe it blank
 
 # Fixed variables
 tmpfile=/tmp/$RANDOM
 
-# Get archived events from database
-declare -a archived=( $(mysql -N -u $mysqluser --password=$mysqlpass -e "select Id from ${mysqldb}.Events where Archived = 1;") )
+if [[ ! -z ${docker_mysql} ]]; then
+  # Get archived events from database
+  declare -a archived=( $(docker exec -it ${docker_mysql} bash -lc 'mysql -N -h ${mysqlhost} -u ${mysqluser} --password=${mysqlpass} -e "select Id from ${mysqldb}.Events where Archived = 1;"') )
```

Comment:
Don't see the point here. It doesn't matter if mysql runs on localhost or on another remote host. Communication happens over tcp. You can use the same `declare -a archived` line from your `else` statement to retrieve this. The magic in Docker-based ZM installations is probably to read the environment variable of the mysql host.

Reply:
The intention was, to isolate the MySQL Instance on the host (Our System runs on docker with Video Cards and Net Cams).
but i will go back with the part to the old one and create a own Container, to use the script with docker.
So the script shrinks in size and is easier to use. ;-)